Document test failed when using memdb

<!-- Please only use this template for submitting common issues -->

**Description**:
I found [this TC](https://github.com/yorkie-team/yorkie/blob/0871fac98488c7de245f19a395c0be1d0317ac19/test/integration/document_test.go#L814) fails when using memdb, but it passes when using mongoDB. (`main` branch)
CI cannot detect it because its using mongoDB when testing.


```
--- FAIL: TestDocumentWithProjects (0.65s)
    --- FAIL: TestDocumentWithProjects/includeSnapshot_test (0.01s)
panic: runtime error: index out of range [0] with length 0 [recovered]
        panic: runtime error: index out of range [0] with length 0

goroutine 4213 [running]:
testing.tRunner.func1.2({0x101ec9e00, 0xc002597f50})
        /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1632 +0x2c4
testing.tRunner.func1()
        /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1635 +0x47c
panic({0x101ec9e00?, 0xc002597f50?})
        /opt/homebrew/Cellar/go/1.23.0/libexec/src/runtime/panic.go:785 +0x124
github.com/yorkie-team/yorkie/test/integration.TestDocumentWithProjects.func3(0xc0025a6340)
        /Users/user/Documents/yorkie/test/integration/document_test.go:828 +0x3b8
testing.tRunner(0xc0025a6340, 0xc003903500)
        /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1690 +0x188
created by testing.(*T).Run in goroutine 4062
        /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1743 +0x5e4
FAIL    github.com/yorkie-team/yorkie/test/integration  26.292s
FAIL
make: *** [test] Error 1
```
**Why**:

--- a/server/backend/database/memory/database.go
+++ b/server/backend/database/memory/database.go
@@ -1564,7 +1564,13 @@ func (d *DB) FindDocInfosByPaging(
 	var docInfos []*database.DocInfo
 	for raw := iterator.Next(); raw != nil; raw = iterator.Next() {
 		info := raw.(*database.DocInfo)
-		if len(docInfos) >= paging.PageSize || info.ProjectID != projectID {
+		// NOTE(raararaara): Unlike MongoDB, which treats PageSize == 0 as "no limit",
+		// memDB requires explicit handling. If PageSize == 0, do not apply any limit.
+		if paging.PageSize > 0 && len(docInfos) >= paging.PageSize {
+			break
+		}
+
+		if info.ProjectID != projectID {
 			break
 		}
 

